src/App.jsx

```diff
@@ -14,33 +15,90 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const [goods, setGoods] = useState(goodsFromServer);
+  const [sortType, setSortType] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const sortGoods = (type, reverse = isReversed) => {
+    let sortedGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+    switch (type) {
+      case 'alphabetically':
+        sortedGoods.sort((a, b) => a.localeCompare(b));
+        break;
+      case 'length':
+        sortedGoods.sort((a, b) => a.length - b.length);
+        break;
+      case 'reset':
+        sortedGoods = [...goodsFromServer];
+        break;
+      default:
+        break;
+    }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+    if (reverse) {
+      sortedGoods.reverse();
+    }
+
+    setGoods(sortedGoods);
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const handleSort = type => {
```

Comment:
Your approach works

But it's a bit more difficult to read and maintain than another one

you could create a common variable for sorted goods (instead of a separate state), schematically:

```
const sortedGoods = handleSort(sortType, isReversed); // where sortType and isReversed are state values
```

On buttons click you just change sortType of isReversed state. Since it is a state values component will be re-rendered on change and the sortedGoods variable will be calculated again